### PR TITLE
cpu/gpu mean op and its unit test

### PR DESCRIPTION
**Pull request by gangliao**
Review after #3133

### CODE REVIEW

File: paddle/operators/mean_op.h [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+template <typename Place, typename T>
+class MeanKernel : public OpKernel {
+public:
+  void Compute(const ExecutionContext& context) const override {
+    auto input = context.Input<Tensor>(0);
+    auto output = context.Output<Tensor>(0);
+
+    output->mutable_data<T>(context.GetPlace());
+
+    EigenScalar<T>::From(*output).device(*(context.GetEigenDevice<Place>())) =
```
Comment (gangliao) [inline paddle/operators/mean_op.h]: Tensor2Eigen still not easy.

File: paddle/operators/mean_op.cc [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+                   "Input/Output of MeanOp must be initialized.");
+    ctx.Output<Tensor>(0)->Resize(framework::make_ddim({1}));
+  }
+};
+
+class MeanOpMaker : public OpProtoAndCheckerMaker {
+public:
+  MeanOpMaker(OpProto *proto, OpAttrChecker *op_checker)
+      : OpProtoAndCheckerMaker(proto, op_checker) {
+    AddInput("X", "The input of mean op");
```
Comment (dzhwinter) [inline paddle/operators/mean_op.cc]: maybe we need a `dim` parameter to indicate which axis will be computed.

File: python/paddle/v2/framework/tests/test_mean_op.py [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+class TestMeanOp(unittest.TestCase):
+    __metaclass__ = OpTestMeta
+
+    def setUp(self):
+        self.type = "mean"
+        self.X = np.random.random((32, 784)).astype("float32")
+        self.Out = np.mean(self.X)
+
+
```
Comment (Superjom) [inline python/paddle/v2/framework/tests/test_mean_op.py]: no real test here ? 

check if  mean_op's output equals inputs's mean

Comment (gangliao) [inline python/paddle/v2/framework/tests/test_mean_op.py]: It looks it already did where OpTestMeta took this job.

Comment (Superjom) [inline python/paddle/v2/framework/tests/test_mean_op.py]: get it.

File: paddle/operators/mean_op.h (hunk 1 above)
Comment (gangliao) [inline paddle/operators/mean_op.h]: Maybe the following code can reach the same goal.

```
tensormap(*output, 0).device(context) = tensormap(*input, 1).mean();
```

File: paddle/operators/mean_op.cc (hunk 2 above)
Comment (gangliao) [inline paddle/operators/mean_op.cc]: I got, thanks.

Comment (gangliao) [inline paddle/operators/mean_op.cc]: discussed with @reyoung  Currently, we only need all mean operator.

**Pull request merged**